src/doc.cpp

```diff
@@ -1538,16 +1538,19 @@ int Doc::PrepareLyricsEnd(FunctorParams *functorParams)
 {
     PrepareLyricsParams *params = dynamic_cast<PrepareLyricsParams *>(functorParams);
     assert(params);
-
-    if ((params->m_currentSyl && params->m_lastNote) && (params->m_currentSyl->GetStart() != params->m_lastNote)) {
-        params->m_currentSyl->SetEnd(params->m_lastNote);
+    auto params_m_currentSyl = params->m_currentSyl;
+    if (!params_m_currentSyl) {
+        return FUNCTOR_STOP; // early return
+    }
+    if (params->m_lastNote && params_m_currentSyl->GetStart() != params->m_lastNote) {
+        params_m_currentSyl->SetEnd(params->m_lastNote);
     }
     else if (m_options->m_openControlEvents.GetValue()) {
-        if ((params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_i)
-            || (params->m_currentSyl->GetWordpos() == sylLog_WORDPOS_m)) {
+        auto wordpos = params_m_currentSyl->GetWordpos();
```

Comment:
Please change auto to the actual type

Reply:
Hm. Ok, I will do that. Type inference is sweet feature!